src/Invite.js

```diff
@@ -14,13 +15,38 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+import PropTypes from 'prop-types';
 import MatrixClientPeg from './MatrixClientPeg';
 import MultiInviter from './utils/MultiInviter';
+import Modal from './Modal';
+import createRoom from './createRoom';
+import sdk from './';
+import { _t } from './languageHandler';
 
 const emailRegex = /^\S+@\S+\.\S+$/;
 
 const mxidRegex = /^@\S+:\S+$/
 
+export const addressTypes = [
+    'mx', 'email',
+];
+
+// PropType definition for an object describing
+// an address that can be invited to a room (which
+// could be a third party identifier or a matrix ID)
+// along with some additional information about the
+// address / target.
+export const UserAddressType = PropTypes.shape({
```

Comment:
This feels like a strange place for a general "user address" type that might no have anything to do with inviting. Putting it in `AddressTile` seems fine?